invokeai/app/services/download/download_base.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) 2023 Lincoln D. Stein and the InvokeAI Development Team
+"""Model download service."""
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from functools import total_ordering
+from pathlib import Path
+from typing import Callable, List, Optional, Union
+
+from pydantic import BaseModel, Field, PrivateAttr
+from pydantic.networks import AnyHttpUrl
+
+
+class DownloadJobStatus(str, Enum):
+    """State of a download job."""
+
+    WAITING = "waiting"  # not enqueued, will not run
+    RUNNING = "running"  # actively downloading
+    COMPLETED = "completed"  # finished running
+    ERROR = "error"  # terminated with an error message
```

Comment:
I noticed that canceling a job marks it as having an error, and the error is a `DownloadJobCancelledException`. This makes it a bit tricky to provide a good user experience. I think we should have a separate status for canceled.

Reply:
The exception is now replaced with a new CANCELLED event.